libflux/fluxdoc/SPEC.md

```diff
@@ -0,0 +1,284 @@
+
+The heading should be a markdown H1 heading, and should be immediately followed by a blank line comment.
+
+Any text between the heading and the code block will be treated as part of the preamble. The preamble should be separated from the code block by a blank line comment. The preamble may be omitted.
+
+Code blocks should contain valid flux code that readers can copy and paste into the InfluxDB data explorer and run with no modifications. Any sample data used in a code example should come from an `array.from()` call in the example code itself, rather than being represented as a markdown- or html-formatted table.
+
+This section may include any number of code examples.
+
+_Example:_
+
+```
+// # Joining two tables
+// 
+// Run this in the data explorer to see the results!
+//
+// ```
+// import "array"
+//
+// sf_temp = array.from(
+//     rows: [
+//         {_time: 2021-06-01T01:00:00Z, _field: "temp", _value: 70},
+//         {_time: 2021-06-01T02:00:00Z, _field: "temp", _value: 75},
+//         {_time: 2021-06-01T03:00:00Z, _field: "temp", _value: 72},
+//     ],
+// )
+//
+// ny_temp = array.from(
+//     rows: [
+//         {_time: 2021-06-01T01:00:00Z, _field: "temp", _value: 55},
+//         {_time: 2021-06-01T02:00:00Z, _field: "temp", _value: 56},
+//         {_time: 2021-06-01T03:00:00Z, _field: "temp", _value: 57},
+//     ],
+// )
+//
+// join(
+//   tables: {sf: sf_temp, ny: ny_temp},
+//   on: ["_time", "_field"]
+// )
+// ```
+```
+
+### Other Information
+
+There are other details about Flux functions that we will want to document, like the return type, the types of the parameters, and whether or not a parameter is required. We should be able to get all of that information from the function signature, and therefore shouldn't need to include it in the doc comment.
+
+### Full Example
+
+For some examples of what a complete doc comment looks like for a flux function, see the doc comments for `join()` and `aggregateWindow()` in the `universe` package.
+
+## Documenting Packages
+
+### Import path
+
+Doc comments for packages should start with the full import path for the package, wrapped in backticks. Nothing else should be included on this line.
+
+_Example:_
+
+```
+// `experimental/geo`
+```
+
+### Description
+
+Include a brief description of the package. This should read as one or more full setences.
+
+_Example:_
+
+```
+// `math`
+//
+// The Flux math package provides basic constants and mathematical functions.
+package math
+```
+
+## Documenting Flux Options
+
+### Name and Description
+
+The first line of a doc comment for a flux option should start with the name of the option wrapped in backticks, followed by a brief explanation of what the option is, or how it is used.
+
+### Extra Details
+
+Additional notes explaining how users can use the option in their code. This section should follow the same formatting rules as the `Extra Details` section for Flux functions described earlier in this document.
```

Comment:
That's something I thought about, but I couldn't think of a good answer. I think if the mark down includes an unordered list, it may be easy for the parser to confuse that for the parameter list, and that could cause problems.

Maybe this can be solved by using a special character or set of characters to delineate the different sections. I think if we do that, we can include any markdown in this section.

That said, for the sake of keeping things organized, I think we should strongly encourage doc writers to keep code blocks in the `Code Examples` section.